Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -547,6 +547,9 @@ public struct ObjectId: BSONValue, Equatable, CustomStringConvertible, Codable {
     /// This `ObjectId`'s data represented as a `String`.
     public let oid: String
 
+    /// The timestamp used to create this `ObjectId`
+    public let timestamp: UInt32
```

Comment:
Any reason this isn't a read-only computed property? 
```
public var timestamp: UInt32 { return UInt32(bson_oid_get_time_t(&oid_t)) }
```

Reply:
I considered that as well, but we don't store a pointer to the `bson_oid_t`, so we'd need to recreate it from the string in here. I guess it depends whether we prefer to always store a `UInt32`, or to recreate the `bson_oid_t` every time this is called. it's not apparent to me which of those is better. what do you think?

Reply:
I'm trying to remember why we chose to store it as a string. It seems to me that ObjectID's are created enough during an applications lifetime that doing all the extra work here to wrap and unwrap it into the libbson type is likely undesirable overhead. Looking more closely at the implementation, it looks like we should instead be storing the `bson_oid_t` (not a pointer, but just the value type - you won't need a deinit for this) and provide computed properties for whatever we want access to.

Reply:
we do in fact create `ObjectId`s in Swift (see `insertOne` and `insertMany`).
I agree that this change makes sense unless if anyone can remember if there was a good reason why we chose to do it this way. I've opened [SWIFT-268](https://jira.mongodb.org/browse/SWIFT-268) for this. I'm going to merge this as-is, and we can address the change in that ticket.